core/config.go

```diff
@@ -333,6 +335,10 @@ func (c *ChainConfig) GetFeature(num *big.Int, id string) (*ForkFeature, *Fork,
 }
 
 func (c *ChainConfig) HeaderCheck(h *types.Header) error {
+	if (h == &types.Header{}) {
+		return ErrHashEmpty
```

Comment:
Still don't quite get it. Why not just return nil when the hash is empty? Why it is needed to treat it as a separate error?

I think I'm just confused why empty header is even allowed in the first place... This only happens in tests, right?